### ISSUE TITLE
[Multi-ESX] Need to remove DB  after hit failure in "new_db()" to handle "config init" failure

### ISSUE THREAD

**Issue by lipingxue** [vmware/docker-volume-vsphere#1162]
When user run "config init" from "NotConfigured" clean state, it will try to call function "new_db()" if the auth db does not exist. 
However, In new_db() function, it can fail in the middle. See the following code snippet: 
```
 def new_db(self):
        """
        Create brand new DB content at self.db_path. Expects clean slate.
        :returns: None for success , a string (with error) for error
        """
        if not self.conn:
            self.__connect()
        err = self.__create_tables()
        if err:
            return err
        err = self.__create_default_tenant()
        if err:
            return err
        err = self.__create_default_privileges()
        if err:
            return err
        return None
```

We need to remove the DB when the new_db() function fails which can ensure that the configuration goes back to a clean state "NotConfigured".

**Comment by govint** [comment]
Simulate error in creating DB,

 bin/vmdkops_admin.py config init --local
Warning: this feature is EXPERIMENTAL
Creating new DB at /etc/vmware/vmdkops/auth-db
Init failed: Vmgroup _DEFAULT does not exist

 bin/vmdkops_admin.py config init --local
Warning: this feature is EXPERIMENTAL
Config DB  is already initialized. Use 'rm --local' to reset

ls -l /etc/vmware/vmdkops/
total 20
-rw-r--r--    1 root     root         15360 May 16 13:33 auth-db
-rw-r--r--    1 root     root          1257 May 16 11:08 log_config.json

**Comment by govint** [comment]
With the fix (but with the simulated error still in place we see the DB is removed if init fails,

bin/vmdkops_admin.py config init --local
Warning: this feature is EXPERIMENTAL
Creating new DB at /etc/vmware/vmdkops/auth-db
Init failed: Vmgroup _DEFAULT does not exist

bin/vmdkops_admin.py config init --local
Warning: this feature is EXPERIMENTAL
Creating new DB at /etc/vmware/vmdkops/auth-db
Init failed: Vmgroup _DEFAULT does not exist

ls -l /etc/vmware/vmdkops/
total 4
-rw-r--r--    1 root     root          1257 May 16 11:08 log_config.json

**Comment by govint** [comment]
Closed via #1259
